File: app/src/main/java/com/vmenon/mpo/adapter/ShowSearchResultsAdapter.java
package com.vmenon.mpo.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.vmenon.mpo.R;
import com.vmenon.mpo.api.Show;

import java.util.List;

public class ShowSearchResultsAdapter extends
        RecyclerView.Adapter<ShowSearchResultsAdapter.ViewHolder> {

    public interface ShowSelectedListener {
        void onShowSelected(Show show);
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private TextView nameText;
        private ImageView imageView;
        private Show show;

        public ViewHolder(View v) {
            super(v);
            nameText = v.findViewById(R.id.showName);
            imageView = v.findViewById(R.id.showImage);
        }
    }

    private List<Show> shows;
    private ShowSelectedListener listener;

    public ShowSearchResultsAdapter(List<Show> myDataset) {
        shows = myDataset;
    }

    public void setListener(ShowSelectedListener listener) {
        this.listener = listener;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.show_result, parent,
                false);
        final ViewHolder vh = new ViewHolder(v);
        v.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (listener != null && vh.show != null) {
                    listener.onShowSelected(vh.show);
                }
            }
        });

        return vh;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Show show = shows.get(position);
        holder.show = show;
        holder.nameText.setText(show.name);

        Glide.with(holder.itemView.getContext())
                .load(show.artworkUrl)
                .centerCrop()
                .crossFade()
                .into(holder.imageView);
    }

    @Override
    public int getItemCount() {
        return shows.size();
    }
}

File: app/src/main/java/com/vmenon/mpo/adapter/ShowSearchResultsAdapter.kt
package com.vmenon.mpo.adapter

import androidx.recyclerview.widget.RecyclerView

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide
import com.vmenon.mpo.R
import com.vmenon.mpo.model.ShowSearchResultsModel
import kotlinx.android.synthetic.main.show_result.view.*

class ShowSearchResultsAdapter(private val shows: List<ShowSearchResultsModel>) :
    RecyclerView.Adapter<ShowSearchResultsAdapter.ViewHolder>() {
    private var listener: ShowSelectedListener? = null

    interface ShowSelectedListener {
        fun onShowSelected(show: ShowSearchResultsModel)
    }

    class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        val nameText: TextView = v.showName
        val imageView: ImageView = v.showImage
        var show: ShowSearchResultsModel? = null
    }

    fun setListener(listener: ShowSelectedListener) {
        this.listener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(
            R.layout.show_result, parent,
            false
        )
        val vh = ViewHolder(v)
        v.setOnClickListener {
            vh.show?.let {
                listener?.onShowSelected(it)
            }
        }

        return vh
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val show = shows[position]
        holder.show = show
        holder.nameText.text = show.show.name

        Glide.with(holder.itemView.context)
            .load(show.show.artworkUrl)
            .centerCrop()
            .crossFade()
            .into(holder.imageView)
    }

    override fun getItemCount(): Int {
        return shows.size
    }
}
